crd/src/util.rs

```diff
@@ -51,18 +65,28 @@ pub async fn get_zk_connection_info(
     Ok(ZookeeperConnectionInformation { connection_string })
 }
 
-// Checks if the provided chroot is legal to be used as a znode according to:
-// https://zookeeper.apache.org/doc/current/zookeeperProgrammers.html#ch_zkDataModel
-//
-// Additionally, if the chroot starts or ends with a '/' these are stripped
-fn check_chroot(chroot: Option<&str>) -> ZookeeperOperatorResult<Option<&str>> {
+/// Check if a string is a valid chroot string to use in a ZooKeeper connection
+/// string
+///
+/// Leading and trailing slashes are ignored, as we strip these before using the
+/// string in a ZooKeeper chroot string.
+///
+/// Slashes elsewhere in the String are allowed as this signifies a valid path.
+///
+/// # Arguments
+///
+/// * `chroot` - The chroot string which should be checked against ZooKeepers naming rules
```

Comment:
```suggestion
/// * `chroot` - The chroot string which should be checked against ZooKeeper's naming rules
```

I'm not good at grammar, might be wrong